src/Shared/Targets/Wrappers/AspNetBufferingTargetWrapper.cs

```diff
@@ -0,0 +1,305 @@
+    /// </p>
+    /// <code lang="C#" source="examples/targets/Configuration API/ASPNetBufferingWrapper/Global.asax.cs" />
+    /// <p>
+    /// Fully working C# project can be found in the <c>Examples/Targets/Configuration API/ASPNetBufferingWrapper</c>
+    /// directory along with usage instructions.
+    /// </p>
+    /// </example>
+    [Target("AspNetBufferingWrapper", IsWrapper = true)]
+    public class AspNetBufferingTargetWrapper : WrapperTargetBase
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AspNetBufferingTargetWrapper" /> class.
+        /// </summary>
+        public AspNetBufferingTargetWrapper()
+            : this(null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AspNetBufferingTargetWrapper" /> class.
+        /// </summary>
+        /// <param name="wrappedTarget">The wrapped target.</param>
+        public AspNetBufferingTargetWrapper(Target wrappedTarget)
+            : this(wrappedTarget, 100)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AspNetBufferingTargetWrapper" /> class.
+        /// </summary>
+        /// <param name="wrappedTarget">The wrapped target.</param>
+        /// <param name="bufferSize">Size of the buffer.</param>
+        public AspNetBufferingTargetWrapper(Target wrappedTarget, int bufferSize)
+        {
+        
+            WrappedTarget      = wrappedTarget;
+            BufferSize         = bufferSize;
+            GrowBufferAsNeeded = true;
+        }
+
+        /// <summary>
+        /// Gets or sets the number of log events to be buffered.
+        /// </summary>
+        /// <docgen category='Buffering Options' order='100' />
+        [DefaultValue(100)]
+        public int BufferSize { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether buffer should grow as needed.
+        /// </summary>
+        /// <value>A value of <c>true</c> if buffer should grow as needed; otherwise, <c>false</c>.</value>
+        /// <remarks>
+        /// Value of <c>true</c> causes the buffer to expand until <see cref="BufferGrowLimit"/> is hit,
+        /// <c>false</c> causes the buffer to never expand and lose the earliest entries in case of overflow.
+        /// </remarks>
+        /// <docgen category='Buffering Options' order='100' />
+        [DefaultValue(false)]
+        public bool GrowBufferAsNeeded { get; set; }
+
+        /// <summary>
+        /// Limits the amount of slots that the buffer should grow
+        /// </summary>
+        private int _bufferGrowLimit;
+
+        /// <summary>
+        /// Gets or sets the maximum number of log events that the buffer can keep.
+        /// </summary>
+        /// <docgen category='Buffering Options' order='100' />
+        public int BufferGrowLimit
+        {
+            get => _bufferGrowLimit;
```

Comment:
Can this be changed to:
```c#
            get
            {
                return growLimit;
            }
```
I know this is not optimal, but we need to reduce the changes to make the rename automatic